### PR TITLE
update docstrings for operation result class

### PR DESCRIPTION
**Pull request by mshaban-msft**
solves #17807

### CODE REVIEW

**Comment by mohamedshabanofficial** [comment]
@maririos 

went through the following classes
    - AnalyzeHealthcareEntitiesOperation
    - AnalyzeHealthcareEntitiesOptions
    - AnalyzeHealthcareEntitiesResult
    - AnalyzeHealthcareEntitiesResultCollection

    - HealthcareEntity
    - HealthcareEntityAssertion
    - HealthcareEntityRelation
    - HealthcareEntityRelationRole
    - HealthcareEntityRelationType

skipped internal and private attributes/methods/constructors

**Review by maririos — COMMENTED** [review]
Great! I found a couple more that need to improve
- HealthcareEntity
  - The class docstring could be improved. See as examples, `CategorizedEntity`, or `LinkedEntity` 
- EntityDataSource
  - The class docstring 

This will cover all the work in Healthcare :) You could leave this PR with that focus and create another one for the `AnalyzeActions` functionality

File: sdk/textanalytics/Azure.AI.TextAnalytics/src/HealthcareEntity.cs [hunk 1]
```diff
@@ -7,7 +7,10 @@
 namespace Azure.AI.TextAnalytics
 {
     /// <summary>
-    /// Health care entity class.
+    /// A model which contains info about the detected health care entity.
```
Comment (maririos) [inline sdk/textanalytics/Azure.AI.TextAnalytics/src/HealthcareEntity.cs:10]:
```suggestion
    /// A model which contains information about the detected healthcare entity.
```

File: sdk/textanalytics/Azure.AI.TextAnalytics/src/HealthcareEntity.cs [hunk 2]
```diff
@@ -7,7 +7,10 @@
...
+    /// A healthcare entity represented as a node in a directed graph
+    /// where the edges are a particular type of relationship between
+    /// the source and target nodes.
```
Comment (maririos) [inline sdk/textanalytics/Azure.AI.TextAnalytics/src/HealthcareEntity.cs:13]: I see JS has this, but I think this only makes sense in the context of the relationships. Maybe we should just remove it

**Review by maririos — APPROVED** [review]

**Pull request merged**